### DISCUSSION TITLE
abrupt in code

### DISCUSSION THREAD

**Discussion by tangyu-ty** [pyopencl/pyopencl#644] [Q&A]
![image](https://user-images.githubusercontent.com/57243868/196940228-4955bc2e-2002-4062-957f-1dfe03947df5.png)
Why is it so abrupt in the middle of the code？（empty = Array）

demo_array.py in examps

**Comment by kif** [comment]
I guess it is just a way to declare empty (which is a factory to generate an empty array) with the same API as the constructor of the class ?

**Comment by tangyu-ty** [comment]
> 

sorry， it's in  pyopencl/array.py

I just wonder why it is placed in the middle of some methods of this module instead of the beginning or the end。'Array' is a class。

class Array is between 338 to 2294. （Number of rows）
Next are some methods or class of array.py

**Comment by tangyu-ty** [comment]
> I guess it is just a way to declare empty (which is a factory to generate an empty array) with the same API as the constructor of the class ?

![image](https://user-images.githubusercontent.com/57243868/196943227-6824c3b0-da8f-46c5-bbee-64620f6146f1.png)
